channeldb/payment_status_test.go

```diff
@@ -0,0 +1,247 @@
+package channeldb
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/stretchr/testify/require"
+)
+
+// TestDecidePaymentStatus checks that given a set of HTLC and a failure
+// reason, the payment's current status is returned as expected.
+func TestDecidePaymentStatus(t *testing.T) {
+	t.Parallel()
+
+	// Create two attempts used for testing.
+	inflight := HTLCAttempt{}
+	settled := HTLCAttempt{
+		Settle: &HTLCSettleInfo{Preimage: lntypes.Preimage{}},
+	}
+	failed := HTLCAttempt{
+		Failure: &HTLCFailInfo{FailureSourceIndex: 1},
+	}
+
+	reason := FailureReasonNoRoute
+	failure := &reason
```

Comment:
yeah this unfortunately is a design flaw here - we want `failure` to be a pointer here to express the case where the is no failure reason. If we don't use a pointer, then this would be a `const`, and the default value 0 is defined as `FailureReasonTimeout`. Updated the docs to clarity it a bit.